### PR TITLE
New PR template for changelog and release notes generation

### PR DESCRIPTION
**Pull request by aarnq**
And why not put it here so you all can see it in action.

### What kind of PR is this?

**Required**: Mark one of the following that is applicable:

- [x] kind/feature       
- [ ] kind/improvement   
- [ ] kind/deprecation   
- [ ] kind/documentation 
- [ ] kind/clean-up      
- [ ] kind/bug           

*Optional*: Mark one or more of the following that are applicable:

> [!important]
> Critical security fixes should be marked with `kind/security`
> Breaking changes should be marked `kind/admin-change` or `kind/dev-change` depending on type

- [ ] kind/admin-change  
- [ ] kind/dev-change    
- [ ] kind/security      
- [ ] [kind/adr]()       

### What does this PR do / why do we need this PR?

Adds a structured PR template with a few mandatory sections and checkboxes that will be used to generate both changelogs (platform administrator facing) and release notes (application developer facing).

#### Checklist

- [x] Proper commit message prefix on all commits
  
- Change checks:
  - [x] The change is transparent
  - [ ] The change is disruptive
  - [x] The change requires no migration steps
  - [ ] The change requires migration steps
- Metrics checks:
  - [ ] The metrics are still exposed and present in Grafana after the change
  - [ ] The metrics names didn't change (Grafana dashboards and Prometheus alerts are not affected)
  - [ ] The metrics names did change (Grafana dashboards and Prometheus alerts were fixed)
- Logs checks:
  - [ ] The logs do not show any errors after the change
- Network Policy checks:
  - [ ] Any changed pod is covered by Network Policies
  - [ ] The change does not cause any dropped packages in the `NetworkPolicy Dashboard`
- Pod Security Policy checks:
  - [ ] Any changed pod is covered by Pod Security Admission
  - [ ] Any changed pod is covered by Gatekeeper Pod Security Policies
  - [ ] The change does not cause any pods to be blocked by Pod Security Admission or Policies
- Falco checks:
  - [ ] The change does not cause any alerts to be generated by Falco

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,62 +1,87 @@
...
+<!-- Choose you PR title carefully as it will be used as the entry in the changelog! -->
 
-<!-- use the format `fixes #<issue number>(, fixes #<issue_number>, ...)` to automatically close the issue when PR gets merged-->
-**Which issue this PR fixes:** fixes #
+### What kind of PR is this?
 
-**Public facing documentation PR** *(if applicable)*
-<!-- https://github.com/elastisys/compliantkubernetes/pull/ -->
+**Required**: Mark one of the following that is applicable:
+- [ ] kind/feature       <!-- This PR adds a new feature -->
```
Comment (crssnd) [inline .github/pull_request_template.md:6]:
```suggestion
**Required**: Mark one of the following that is applicable:
- [ ] BREAKING CHANGE
- [ ] kind/feature       <!-- This PR adds a new feature -->
```

I am thinking to better highlight changes that could break user applications

Comment (aarnq) [reply .github/pull_request_template.md:6]: That would be what `kind/dev-change` is for, and similarly `kind/admin-change` for platform admins. With one of those checked the additional sections "Application Developer notice" respectively "Platform Administrator notice" becomes mandatory.

Comment (davidumea) [reply .github/pull_request_template.md:6]: :thinking: That was not super clear to me either

Maybe this
> Optional: Mark one or more of the following that are applicable:

could be changed to clarify that

Comment (aarnq) [reply .github/pull_request_template.md:6]: Would this improve the situation?

---

*Optional*: Mark one or more of the following that are applicable:

> [!important]
> Breaking changes must be marked `kind/admin-change` or `kind/dev-change` depending on type
> Critical security fixes must be marked with `kind/security`

- [ ] kind/admin-change  
- [ ] kind/dev-change    
- [ ] kind/security      
- [ ] [kind/adr]()

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -1,62 +1,87 @@
...
+    - bin: changes to management binaries or scripts
+    - config: changes to configuration
+    - docs: changes to documentation
+    - tests: changes to tests
+    - pipeline: changes to the pipeline
+    - release: release related
+  -->
+- Change checks:
+  - [ ] Change is transparent and requires no migration steps
+  - [ ] Change is disruptive and requires migration steps
```
Comment (Xartos) [inline .github/pull_request_template.md:72]: Do we want a middle step that requires some migration but isn't disruptive?

```suggestion
  - [ ] Change is transparent and requires no migration steps
  - [ ] Change is transparent but requires some migrations steps
  - [ ] Change is disruptive and requires migration steps
```

File: .github/pull_request_template.md (hunk 1 above)
Comment (davidumea) [reply .github/pull_request_template.md:6]: I like that :+1:

Comment (crssnd) [reply .github/pull_request_template.md:6]: perfekt!

**Review by crssnd — APPROVED** [review]

**Review by davidumea — APPROVED** [review]

**Review by Xartos — APPROVED** [review]

**Pull request merged**